torch/csrc/jit/codegen/cuda/lower_misaligned_vectorization.cpp

```diff
@@ -173,7 +180,7 @@ class MisalignedVectorizationModifier : public kir::ExprMutator {
 
     // >>>>>>>>>>>>>
     auto last_root_domain_index = createNamedScalarFromValue(
-        parent_scope_ite->thenBody(), indices.back(), "last_root_domain_index");
+        parent_scope_ite->thenBody(), last_index, "last_root_domain_index");
```

Comment:
BTW: I don't think this is correct. What if the last dim is a broadcast? But my PR does not make this worse.

Reply:
Don't think it's important to fix it now, but can you please leave an inline comment or file a github issue so that we don't forget it?

Reply:
left a TODO